waffle_hub/utils/memory.py

```diff
@@ -1,18 +1,15 @@
 import torch
 
 
-def device_context(device):
-    """For gpu memory decorator"""
-
-    def decorator(func):
-        def wrapper(*args, **kwargs):
-            if torch.cuda.is_available():
-                torch.cuda.init()  # for thread
-            func(*args, **kwargs)
-            if torch.cuda.is_available() and device != "cpu":
-                # Memory free
-                torch.cuda.empty_cache()
-
-        return wrapper
-
-    return decorator
+def device_context(func):
+    def wrapper(*args, **kwargs):
+        result = func(*args, **kwargs)
```

Comment:
Don't we need to clear cache before `func` do something?

Reply:
All functions clear the cache at the end, so I think the cache is already cleared before the function works.

Because we don't have to think about threading anymore, we're responsible for whether a single function works well.
In addition, If another user runs our module to multi-process, the memory management is up to that user.

Reply:
This function can be independently used for various use cases, not limited to the `waffle_hub` workflow as we discussed before.

For instance,
```python
# waffle_menu
results = some_sampling.sample(..., "cuda")

# some pytorch jobs
...

# waffle_menu
results = something_else(..., "cuda")
```

The third job will function alongside some garbage cahce.

Reply:
I modified it.

But we need to rethink whether we need to help with not clearing the cache in other tasks.
We should also consider whether the memory-related function should be in `waffle_hub` or transferred to `waffle_utils`.
@snuailab/waffle @snuailab/waffle_reviewer 

I accepted the opinion because I think clearing the cache before and after `func` guarantees our method to operate in a normal environment.

Reply:
I agree. But, it will cause pytorch dependency to `waffle_util`. Let's discuss about it later.